### PR TITLE
[No JIRA] Add code review guidelines

### PR DESCRIPTION
**Pull request by k0nserv**
+ [x] Check this if you have read and followed the [Contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: CODE_REVIEW_GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Code Review Guidelines
+
+Backpack is Skyscanner's Design System. It should be an encoding of the Skyscanner visual design language, patterns, and ideas.
+
+Backpack should be the best way to build and design Skyscanner UI.
+
+## Anchoring in Design
+Backpack starts with design and all changes or additions should be previously agreed on in the wider design community before being implemented in Backpack. In absence of having gone through design review changes should have been reviewed and approved by a Backpack designer.
```
Comment (georgegillams) [inline CODE_REVIEW_GUIDELINES.md]:
```suggestion
Backpack starts with design, and all changes or additions should be previously agreed on in the wider design community before being implemented in Backpack. In absence of having gone through design review changes should have been reviewed and approved by a Backpack designer.
```

File: CODE_REVIEW_GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Code Review Guidelines
+
+Backpack is Skyscanner's Design System. It should be an encoding of the Skyscanner visual design language, patterns, and ideas.
+
+Backpack should be the best way to build and design Skyscanner UI.
```
Comment (georgegillams) [inline CODE_REVIEW_GUIDELINES.md]:
```suggestion
Backpack _should be_ the best way to build and design Skyscanner UI.
```

File: CODE_REVIEW_GUIDELINES.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Anchoring in Design
+Backpack starts with design and all changes or additions should be previously agreed on in the wider design community before being implemented in Backpack. In absence of having gone through design review changes should have been reviewed and approved by a Backpack designer.
+
+## Quality
+Backpack is inteded to reduce work when implementing Skyscanner UI and should be of high quality. It's expected that implementing UI as part of Backpack will take more time and effort than the equivalent work outside of Backpack.
```
Comment (georgegillams) [inline CODE_REVIEW_GUIDELINES.md]:
```suggestion
Backpack aims to reduce work when implementing Skyscanner UI and should be of high quality. It's expected that implementing UI as part of Backpack will take more time and effort than the equivalent work outside of Backpack.
```

Comment (k0nserv) [reply CODE_REVIEW_GUIDELINES.md]: I like "is intended" better 😄

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<!--
+Thanks for contributing to Backpack :pray:
+
+Please include a description of the changes you are introducing and if appropriate some screenshots.
```
Comment (shaundon) [inline .github/pull_request_template.md]:
```suggestion
Please include a description of the changes you are introducing and some screenshots if appropriate.
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+-->
+
++ [ ] Check this if you have read and followed the [Contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)
+
+
+_If you are curious about how we review please read through the [Code Review Gudielines](./CODE_REVIEW_GUIDELINES.md)_
```
Comment (shaundon) [inline .github/pull_request_template.md]:
```suggestion
_If you are curious about how we review, please read through the [Code Review Guidelines](./CODE_REVIEW_GUIDELINES.md)_
```

File: CODE_REVIEW_GUIDELINES.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Code Review Guidelines
```
Comment (shaundon) [inline CODE_REVIEW_GUIDELINES.md]:
```suggestion
# Code review guidelines
```
We use sentence case for headings in `CONTRIBUTING.md` so this matches that.

File: CODE_REVIEW_GUIDELINES.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+Backpack is Skyscanner's Design System. It should be an encoding of the Skyscanner visual design language, patterns, and ideas.
+
+Backpack should be the *best* way to build and design Skyscanner UI.
+
+## Anchoring in Design
```
Comment (shaundon) [inline CODE_REVIEW_GUIDELINES.md]:
```suggestion
## Anchoring in design
```

File: CODE_REVIEW_GUIDELINES.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+Backpack starts with design, and all changes or additions should be previously agreed on in the wider design community before being implemented in Backpack. In absence of having gone through design review changes should have been reviewed and approved by a Backpack designer.
```
Comment (shaundon) [inline CODE_REVIEW_GUIDELINES.md]:
```suggestion
Backpack starts with design, and all changes or additions should be previously agreed on in the wider design community before being implemented in Backpack. In absence of having gone through design review, changes should have been reviewed and approved by a Backpack designer.
```

File: CODE_REVIEW_GUIDELINES.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+## Quality
+
+Backpack is inteded to reduce work when implementing Skyscanner UI and should be of high quality. While it's expected that implementing UI as part of Backpack will take more time and effort than the equivalent work outside of Backpack, the results empower everyone to go faster.
+
+All changes to Backpack should
+
++ Follow the Backpack style-guide.
```
Comment (shaundon) [inline CODE_REVIEW_GUIDELINES.md]:
```suggestion
+ Follow the Backpack style guide.
```

Is there something we can link to here?

File: CODE_REVIEW_GUIDELINES.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
++ Pass all linting checks.
++ Have zero warnings.
++ Expose a _delightful_ API surface.
++ Have high test coverage.
++ Match the design.
++ Have considered accessibility and through the API encourage consumers to build accessible products. e.g. Making `accessibilityLabel` required rather than optional.
++ Support theming when appropriate.
++ Feature good examples.
++ Follow [SemVer](https://semver.org/) erring on the side of one too many breaking changes rather than shipping something breaking as a minor or patch.
++ Have considered extreme cases such as user with larger text or smaller devices.
```
Comment (shaundon) [inline CODE_REVIEW_GUIDELINES.md]:
```suggestion
+ Have considered extreme cases such as users with larger text or smaller devices.
```

File: CODE_REVIEW_GUIDELINES.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
++ Be well documented.
++ Be closed for modificaiton of look and feel.
```
Comment (shaundon) [inline CODE_REVIEW_GUIDELINES.md]:
```suggestion
+ Be closed for modification of look and feel.
```

File: .github/pull_request_template.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+<!--
+Thanks for contributing to Backpack :pray:
+
+Please include a description of the changes you are introducing and if appropriate some screenshots.
+-->
+
++ [ ] Check this if you have read and followed the [Contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)
```
Comment (shaundon) [inline .github/pull_request_template.md]:
```suggestion
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -8,6 +8,10 @@ Please follow the [New York Times Objective-C style-guide](https://github.com/NY
 
 Wrap all Objective-C in `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END` blocks. Make sure to annotate any types that are nullable correctly.
 
+## How we review Backpack contributions
+
+Please see the [Code Review Guidelines](CODE_REVIEW_GUIDELINES.md).
```
Comment (shaundon) [inline CONTRIBUTING.md]:
```suggestion
Please see the [code review guidelines](CODE_REVIEW_GUIDELINES.md).
```

File: CODE_REVIEW_GUIDELINES.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+## Quality
+
+Backpack is inteded to reduce work when implementing Skyscanner UI and should be of high quality. While it's expected that implementing UI as part of Backpack will take more time and effort than the equivalent work outside of Backpack, the results empower everyone to go faster.
+
+All changes to Backpack should
+
++ Follow the Backpack style guide.
++ Pass all linting checks.
++ Have zero warnings.
++ Expose a _delightful_ API surface.
```
Comment (Szabadember) [inline CODE_REVIEW_GUIDELINES.md]: What is a delightful API surface?

File: CODE_REVIEW_GUIDELINES.md [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
++ Have high test coverage.
++ Match the design.
++ Have considered accessibility and through the API encourage consumers to build accessible products. e.g. Making `accessibilityLabel` required rather than optional.
++ Support theming when appropriate.
++ Feature good examples.
++ Follow [SemVer](https://semver.org/) erring on the side of one too many breaking changes rather than shipping something breaking as a minor or patch.
++ Have considered extreme cases such as users with larger text or smaller devices.
++ Be well documented.
++ Be closed for modification of look and feel.
```
Comment (Szabadember) [inline CODE_REVIEW_GUIDELINES.md]: Maybe we could add "Check Unreleased.md" or add to CONTRIBUTING.md that Unreleased.md should be filled?

Comment (k0nserv) [reply CODE_REVIEW_GUIDELINES.md]: Yeah good shout

File: CODE_REVIEW_GUIDELINES.md (hunk 14 above)
Comment (k0nserv) [reply CODE_REVIEW_GUIDELINES.md]: Open to interpration because it's very complex to define

**Pull request merged**